core/model/message.go

```diff
@@ -79,3 +79,41 @@ type MessagesStats struct {
 	Read       *int64 `json:"read_count" bson:"read_count"`
 	Unread     *int64 `json:"not_read_count" bson:"not_read_count"`
 }
+
+///
+
+//InputMessage is passed by the adapters for creating a message in the core module
+type InputMessage struct {
+	OrgID string `json:"org_id"`
+	AppID string `json:"app_id"`
+
+	Priority int               `json:"priority"`
+	Subject  string            `json:"subject"`
+	Sender   *InputSender      `json:"sender"`
+	Body     string            `json:"body"`
+	Data     map[string]string `json:"data"`
+
+	//recipients related
+	Recipients             []InputMessageRecipient  `json:"recipients"`
+	RecipientsCriteriaList []InputRecipientCriteria `json:"recipients_criteria_list"`
+	Topic                  *string                  `json:"topic"`
+}
+
+// InputSender is passed by the adapters for creating a message in the core module
+type InputSender struct {
+	UserID string `json:"user_id"`
+	Name   string `json:"name"`
+}
+
+// InputMessageRecipient is passed by the adapters for creating a message in the core module
+type InputMessageRecipient struct {
+	UserID string `json:"user_id"`
+	Name   string `json:"name"`
```

Comment:
Recipient:
- I removed it as I also think it is not used

Sender:
I am not sure we can remove it wihtout to break something. Most probably this is shown in the client app and some BBs send notifications(and set the name as they know it) via the Notificatiosns BBs. The user record in the Notifications BB is missing. @pmarkhennessy ?

Reply:
Hi @mdryankov , thanks for your check. You mean that Polls BB sets _user_id_ and _name_ or only _user_id_ and empty _name_?

Reply:
Hi @petyos, I took another look at this with @pmarkhennessy today and we also couldn't find any usage of this aside from the Polls. Even the polls usage does not actually get displayed in the client. 

Having thought about it a bit more, I am actually wondering if we really need the `InputSender` at all. Shouldn't we be able to get the sender information from the token? In other words, if it is a user/admin sending the message then the user's token claims will contain their ID/name. 

If it is a BB sending the notification, then it should be a "system" sender anyway. It looks like the Polls BB is using the  internal/bbs APIs to send a notification on where the sender is set to the user who created the poll, but I actually think that this is misleading and it would be better for the sender to be "system". It may actually make sense to record info about the sender even if it is a "system" sender using the claims from the service account token (ID/name just like for user claims) for audit purposes.

Let me know if you have any thoughts or concerns about removing the input sender and using the claims to populate the sender info in the message instead. Thanks!

Reply:
This sounds very good to me! Thanks!
@shurwit @mdryankov @pmarkhennessy